projects/igniteui-angular/src/lib/avatar/avatar.component.spec.ts

```diff
@@ -29,122 +41,183 @@ describe('Avatar', () => {
     it('Initializes avatar with auto-incremented id', () => {
         const fixture = TestBed.createComponent(InitAvatarComponent);
         fixture.detectChanges();
-        const avatar = fixture.componentInstance.avatar;
-        const domAvatar = fixture.debugElement.query(By.css('igx-avatar')).nativeElement;
+        const instance = fixture.componentInstance.avatar;
+        const hostEl = fixture.debugElement.query(By.css(baseClass)).nativeElement;
+
+        expect(instance.id).toContain('igx-avatar-');
+        expect(hostEl.id).toContain('igx-avatar-');
+
+        instance.id = 'customAvatar';
+        fixture.detectChanges();
+
+        expect(instance.id).toBe('customAvatar');
+        expect(hostEl.id).toBe('customAvatar');
+    });
+
+    it('Initializes square and round avatar', () => {
+        const fixture = TestBed.createComponent(AvatarWithAttribsComponent);
+        fixture.detectChanges();
+        const instance = fixture.componentInstance.avatar;
+        const hostEl = fixture.debugElement.query(By.css(baseClass)).nativeElement;
+
+        expect(instance.roundShape).toBeTruthy();
+        expect(hostEl.classList).toContain(classes.round);
 
-        expect(avatar.id).toContain('igx-avatar-');
-        expect(domAvatar.id).toContain('igx-avatar-');
+        instance.roundShape = false;
+
+        fixture.detectChanges();
+        expect(instance.roundShape).toBeFalsy();
+        expect(hostEl.classList).not.toContain(classes.round);
+    });
+
+    it('Initializes small, medium, and large avatar', () => {
```

Comment:
This test name doesn't correspond to what the test does. If the test is intended to change the size of the existing avatar but not initializing new one, consider changing the name to something like 'Change size attribute'.